ga4gh/backend.py

```diff
@@ -229,8 +245,13 @@ def _getDatasetFromVariantsRequest(self, request):
         return dataset
 
     def _getDatasetFromCompoundId(self, compoundId):
```

Comment:
We're getting a bit of a proliferation of methods to get a dataset, and having ``compoundId`` refer to both a string and an object is very confusing. How about this:

1. As soon as we have an ID that we want to decompose, we allocate a CompoundId instance, and use this object to get attributes we want.
2. Then, use this to get the values we want. For example,

```python
def getVariantSet(self, id_):   
     compoundId = variants.VariantSetCompoundId(id_) # This takes care of the parsing, and throw exception if malformed.
     dataset = self.getDataset(compoundId.dataset)
     return self.runGetRequest(dataset.getVariantSetIdMap(), id_) 

def getVariant(self, id_):
     compoundId =  variants.VariantCompoundId(id_)
     dataset = self.getDataset(compoundId.dataset)
     variantSet = _safeMapQuery(  
          dataset.getVariantSetIdMap(), compoundId.variantSetId)  
      variant = variantSet.getVariant(compoundId)
      return variant.toJsonString()
```
Does this make sense? 

I've just realised that the backend's API really isn't very helpful at the moment. I'm going to attempt a quick refactor to see if it clears things up a bit.